src/modules/sync/modelTransfomers/workflow.ts

```diff
@@ -0,0 +1,47 @@
+import { throwError } from "../../../utils/error.js";
+import { omit } from "../../../utils/object.js";
+import { EnvironmentModel } from "../generateSyncModel.js";
+import { WorkflowSyncModel } from "../types/syncModel.js";
+
+export const transformWorkflowModel = (
+  environmentModel: EnvironmentModel,
+): ReadonlyArray<WorkflowSyncModel> =>
+  environmentModel.workflows.map(workflow => ({
+    ...omit(workflow, ["id", "scheduled_step"]),
+    steps: workflow.steps.map(step => ({
+      ...omit(step, ["id"]),
+      transitions_to: step.transitions_to.map(t => ({
+        step: {
+          codename:
+            [...workflow.steps, workflow.published_step, workflow.archived_step].find(s => s.id === t.step.id)?.codename
```

Comment:
It would be better to create funcitons for steps and scopes, so it is easier to read. But that's more of a nitpick, but I would like that more :D